### PR TITLE
[Bugfix] Fix M-RoPE position calculation when chunked prefill is enabled

### PR DESCRIPTION
**Pull request by imkero**
Fix `MRotaryEmbedding`'s `get_input_positions` when chunked prefill is enabled.

It only slice at the left-hand side of generated `llm_positions` currently (forgetting the right-hand side). This PR add right-hand slice position in it to support chunked prefill.

https://github.com/vllm-project/vllm/blob/1d754726265d52773653e53e1a18f6eb63122480/vllm/model_executor/layers/rotary_embedding.py#L923-L928

## Explanation

To make it more clear, here is an example with following configuration:
- assume a `len=40` prompt
- `enable_chunked_prefill=True`, and `max_num_batched_tokens=32`
- add some log in `model_runner.py::ModelInputForGPUBuilder::build` near
  
   https://github.com/vllm-project/vllm/blob/1d754726265d52773653e53e1a18f6eb63122480/vllm/worker/model_runner.py#L952-L957

Result:

| step | before this fix | after this fix |
| - | - | - |
| 1<sup>st</sup> prefill chunk | <pre>context_lens: [0]<br>query_lens: [32]<br>seq_lens: [32]<br>input_tokens_lens: [40]<br>mrope_input_positions: torch.Size([3, <b>40</b>])</pre> | <pre>context_lens: [0]<br>query_lens: [32]<br>seq_lens: [32]<br>input_tokens_lens: [40]<br>mrope_input_positions: torch.Size([3, <b>32</b>])</pre> |
| 2<sup>nd</sup> prefill chunk | broken in prev step | <pre>context_lens: [32]<br>query_lens: [8]<br>seq_lens: [40]<br>input_tokens_lens: [40]<br>mrope_input_positions: torch.Size([3, 8])</pre> |
| 1<sup>st</sup> decode | broken in prev step | <pre>context_lens: [40]<br>query_lens: [1]<br>seq_lens: [41]<br>input_tokens_lens: [40]<br>mrope_input_positions: torch.Size([3, 1])</pre> |

Related error log:

```
RuntimeError: shape '[40, -1, 128]' is invalid for input of size 49152
```

the error occurs near:

https://github.com/vllm-project/vllm/blob/1d754726265d52773653e53e1a18f6eb63122480/vllm/model_executor/layers/rotary_embedding.py#L807-L825

## About the test I added

1. Qwen2-VL's M-RoPE works only when there are some multi-modal inputs,
   so an image is included in the inputs
2. however, Qwen2-VL currently won't work properly when chunked prefill is enabled and there are some multi-modal inputs (it **assumes the input is never chunked**)

   https://github.com/vllm-project/vllm/blob/1d754726265d52773653e53e1a18f6eb63122480/vllm/model_executor/models/qwen2_vl.py#L1229-L1238

   here use a *hacky* way: provide a **zero-length** image to make it happy

3. and finally we achieved these requirements to allow our test continue
   - chunked prefill enabled
   - M-RoPE works

### CODE REVIEW

**Comment by DarkLight1337** [comment]
@ywang96 I thought chunked prefill isn't supported for VLMs yet?

**Comment by imkero** [comment]
> @ywang96 I thought chunked prefill isn't supported for VLMs yet? Or is this just not tested properly?

@DarkLight1337 I think after #8346 merged, vLLM seems be able to support multi-modal models' chunked prefill? See 

https://github.com/vllm-project/vllm/blob/1d754726265d52773653e53e1a18f6eb63122480/tests/models/decoder_only/audio_language/test_ultravox.py#L22-L27

and also 

https://github.com/vllm-project/vllm/blob/main/vllm/multimodal/base.py#L288-L328

I am using a forked vLLM which added chunked prefill and prefix caching for Qwen2-VL only. And I found there is a MRotaryEmbedding's fault which prevents Qwen2-VL's chunked prefill works properly (trying to fix it in this PR).

It seems some other changes should be made in Qwen2-VL's impl to support chunked prefill, so this PR is a very early fix.

Maybe I can help adding chunked prefill to Qwen2-VL in vLLM's impl?

**Comment by DarkLight1337** [comment]
> > @ywang96 I thought chunked prefill isn't supported for VLMs yet? Or is this just not tested properly?
> 
> @DarkLight1337 I think after #8346 merged, vLLM seems be able to support multi-modal models' chunked prefill? See
> 
> https://github.com/vllm-project/vllm/blob/1d754726265d52773653e53e1a18f6eb63122480/tests/models/decoder_only/audio_language/test_ultravox.py#L22-L27
> 
> and also
> 
> https://github.com/vllm-project/vllm/blob/1d754726265d52773653e53e1a18f6eb63122480/vllm/multimodal/base.py#L288-L328
> 
> I am using a forked vLLM which added chunked prefill and prefix caching for Qwen2-VL only. And I found there is a MRotaryEmbedding's fault which prevents Qwen2-VL's chunked prefill from working properly (trying to fix it in this PR).
> 
> It seems some other changes should be made in Qwen2-VL's impl to support chunked prefill, so this PR is a very early fix.
> 
> Maybe I can help adding chunked prefill to Qwen2-VL as well?

Oh, I confused it with speculative decoding, sorry. Chunked prefill is supported but not implemented for Qwen2-VL yet.

**Comment by DarkLight1337** [comment]
Feel free to open another PR for full support. Meanwhile we can fix M-RoPE using this PR.

**Review by DarkLight1337 — APPROVED** [review]
The code looks reasonable, but let's see whether the tests can pass. Thanks for the detailed explanation!

**Comment by imkero** [comment]
> The code looks reasonable, but let's see whether the tests can pass. Thanks for the detailed explanation!

Please help retry this CI? https://buildkite.com/vllm/fastcheck/builds/8107
It failed because of a network problem.

```
requests.exceptions.ReadTimeout: (ReadTimeoutError("HTTPSConnectionPool(host='huggingface.co', port=443): Read timed out. (read timeout=10)")
```

And please also help unblock `Multi-Modal Models Test (Standard)` to trigger the test added in this PR? Thanks.

**Comment by DarkLight1337** [comment]
Nice work!

**Pull request merged**
